trin-core/src/portalnet/storage.rs

```diff
@@ -462,6 +457,34 @@ impl PortalStorage {
         Ok(())
     }
 
+    /// Public method for evicting a certain content id. Will revert RocksDB deletion if meta_db
+    /// deletion fails.
+    pub fn evict(&self, id: [u8; 32]) -> anyhow::Result<()> {
+        let deleted_value = self.db.get(&id)?;
+        self.db.delete(&id)?;
+        // Revert rocksdb action if there's an error with writing to metadata db
+        if let Err(err) = self.meta_db_remove(&id) {
+            if let Some(value) = deleted_value {
+                self.db_insert(&id, &value)?;
+            }
+            return Err(anyhow!("failed deletion {err}"));
+        }
+        Ok(())
+    }
+
+    /// Public method for looking up a content key by its content id
+    pub fn lookup_content_key(&self, id: [u8; 32]) -> anyhow::Result<Vec<u8>> {
+        let conn = self.sql_connection_pool.get()?;
+        let mut query = conn.prepare(CONTENT_KEY_LOOKUP_QUERY)?;
+        let id = id.to_vec();
+        let mut result = query.query_map([id], |row| Ok(ContentKey { key: row.get(0)? }))?;
+        let result = result
+            .next()
+            .ok_or(anyhow!("Unable to find content id in meta db"))?;
```

Comment:
instead of an error if not found, it would make more sense to return `Option`.

Reply:
I'm not so sure, we really never expect to be unable to lookup a content key. Our metadb and rocksdb are expected to remain in sync, so if for some reason we're unable to find a content key for a given content id, that means something's out of sync, which imo warrants an `Error`.

Reply:
well that assumes that we only look up a content key for a content ID that we have retrieved from the database. if we are given some arbitrary content ID, and we want to look up the content key for some reason, then the caller would not expect an error if the content key was not found.

I submit that here we know that the method will only be called for previously retrieved content IDs, but that may not be true in some possible future use case for the method. I would prefer that we `Option::unwrap` (or emit a warn/error) in the case where the database not holding the content key represents some sort of data corruption.

Reply:
Ok, I've made the suggested change, since imo it's a fairly trivial point to go back and forth about. But, for the record, I don't feel like we should be building for "some possible future use case for the method". Especially without a strong example use case. I was under the impression that content ids are primarily for internal use within storage for management. I can't think of a use case where a public facing method on `PortalStorage` will be looking up a content id directly - without knowledge of the content key? Can you?